### PR TITLE
[CORE-1041] Refactor realEnv code into a separate package

### PR DESCRIPTION
**Pull request by FahadBSyed**

### CODE REVIEW

File: src/server/pfs/fuse/fuse_test.go [hunk 1]
```diff
@@ -579,7 +579,7 @@ func TestMountDir(t *testing.T) {
 		require.Equal(t, "bar", filepath.Base(files[0].Name()))
 		require.Equal(t, "foo", filepath.Base(files[1].Name()))
 
-		//require.ElementsEqualUnderFn(t, []string{"foo", "bar"}, []string{"foo", "bar"}, func(f interface{}) interface{} { return f.(iofs.FileInfo).Name() })
+		// require.ElementsEqualUnderFn(t, []string{"foo", "bar"}, []string{"foo", "bar"}, func(f interface{}) interface{} { return f.(iofs.FileInfo).Name() })
```
Comment (armaanv) [inline src/server/pfs/fuse/fuse_test.go:582]: Is this an auto formatting change from the IDE?

File: src/server/pps/server/pipeline_controller.go [hunk 2]
```diff
@@ -434,10 +434,10 @@ func (pc *pipelineController) apply(ctx context.Context, pi *pps.PipelineInfo, r
 	return nil
 }
 
-// RcIsFresh returns a boolean indicating whether rc has the right labels
+// rcIsFresh returns a boolean indicating whether rc has the right labels
```
Comment (armaanv) [inline src/server/pps/server/pipeline_controller.go:437]: I think this function was being exported before your changes. Do we need to lock it down?

**Review by armaanv — APPROVED** [review]
looks good to me. I left a couple of minor comments

File: src/server/pfs/fuse/fuse_test.go (hunk 1 above)
Comment (FahadBSyed) [reply src/server/pfs/fuse/fuse_test.go:582]: Yeah it is, you are right

File: src/server/pps/server/pipeline_controller.go (hunk 2 above)
Comment (FahadBSyed) [reply src/server/pps/server/pipeline_controller.go:437]: I think it was unexported before based on the diff here:
https://github.com/pachyderm/pachyderm/blob/707a081ac581286d758ba4732aaedd409a926a8c/src/server/pps/server/pipeline_controller.go

**Pull request merged**